SecurityPkg/VariableAuthenticated/SecureBootConfigDxe/SecureBootConfigImpl.c

```diff
@@ -2652,8 +2661,53 @@ UpdateDeletePage (
       //
       // Display GUID and help
       //
-      GuidToString (&Cert->SignatureOwner, GuidStr, 100);
-      GuidID = HiiSetString (PrivateData->HiiHandle, 0, GuidStr, NULL);
+
+
+      // Only half of the buffer size may be used for the names
+      // Because it needs to be resized to 16b which
+      // will cause it to grow in size 2x
+
+      CertificateInfoReadSuccess = X509GetSubjectName(
+        (UINT8*)Cert->SignatureData,
+        (UINTN)CertList->SignatureSize,
+        (UINT8*)CertificateInfoStr,
+        &CertificateInfoStrSizeHalf
+      )
+
+      if(!CertificateInfoReadSuccess)
+      {
+        CertificateInfoReadSuccess = X509GetCommonName(
+          (UINT8*)Cert->SignatureData,
+          (UINTN)CertList->SignatureSize,
+          (UINT8*)CertificateInfoStr,
+          &CertificateInfoStrSizeHalf
+        )
+      }
+
+      CertificateInfoReadSuccess = CertificateInfoReadStatus && (StrLen(CertificateInfoStr) < 2)
+
+      if(!CertificateInfoReadSuccess)
+      {
+        StrCatS(CertificateInfoStr, CertificateInfoStrSize, UNKNOWN_CERT_PLACEHOLDER);
+      }
+
+      // X509 functions get the name in 8b chars, we need to convert it
+      // to 16b chars
+
+      // There is AsciiStrnToUnicodeStrS function to do that but it
+      // can't be used in place and allocating another
+      // AllocateZeroPool(100) fails with EFI_OUT_OF_RESOURCES

```

Comment:
If AllocateZeroPool(100) return EFI_OUT_OF_RESOURCES then everything should simply fall apart. There are plenty of places where AllocateZeroPool is used, so there must be something wrong going on here.

Reply:
Maybe, that was an hypothesis i had because the certificate entries were not wrong, the list was not rendered at all, even the `help` on the right side. I copied the usage of `AllocateZeroPool` from this function, including the GOTO statement and figured that GOTO must be causing the rendering to be skipped. 

Either way it works fine now. Should I try to rewrite this using AsciiStrnToUnicodeStrS and a second buffer instead?

Reply:
@philipandag Yes, let's try that